src/robot_interface/models/mission/task.py

```diff
@@ -79,6 +81,7 @@ class DockingProcedure(MotionTask):
 
     behavior: Literal["dock, undock"]
     name: Literal["docking_procedure"] = "docking_procedure"
+    id: UUID = field(default_factory=uuid4)
```

Comment:
Instead of setting `id` for all step classes, you can add `id: UUID = field(default_factory=uuid4, init=false)` in the base class

Reply:
Good catch :+1: